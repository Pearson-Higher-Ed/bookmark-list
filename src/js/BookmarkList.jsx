import React, {PropTypes} from 'react';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import SvgIcon from 'material-ui/SvgIcon';

class BookmarkList extends React.Component {
  constructor(props) {
    super(props);
    this.handleModalOpen=this.handleModalOpen.bind(this);
    this.state = { 
      bookmarkList: this.props.bookmarksArr,
      modalOpen: false,
      bookmarkId:''
    };
  }

  handleModalOpen (id) {
    this.setState({
      modalOpen: true,
      bookmarkId:id
    });    
  };

  handleModalClose = () =>  {
    this.setState({modalOpen: false});
  };

  handleClick(uri, e) {
    this.props.clickBookmarkHandler(uri);
    const listDom = Array.from(document.getElementsByClassName('o-bookmark-section'));
    listDom.map((node) => node.className = 'o-bookmark-section');
    e.target.parentNode.classList.add('focused');
  }

  handleRemoveBookmark(id) {
    this.props.removeBookmarkHandler(id);
    const index = this.props.bookmarksArr.findIndex(item => item.id === id);
    this.props.bookmarksArr.splice(index, 1);
    this.setState({
      bookmarkList: this.props.bookmarksArr,
      modalOpen: false
    });
  }

  renderNoBookmarks() {    
    return (<div className="o-bookmark-empty-help">
      <div tabIndex="0" className="o-bookmark-empty-message">
        <p>No Bookmarks were found.</p>
      </div>
    </div>);
  }

  onFocus(e) {
    e.target.parentNode.classList.add('focused');
    return true;
  }

  onLiBlur(e) {
    if (e.shiftKey && e.keyCode === 9) {
      e.target.parentNode.classList.remove('focused');
    }
  }

  onBlur(e) {
    e.target.parentNode.className = ' o-bookmark-section';
    return true;
  }

  renderBookmarks() {
    const that = this;
    // const {formatMessage} = this.props.intl;
    // const {formatDate} = this.props.intl;
    // const {formatTime} = this.props.intl;
    const DialogStyle = {
      dialogContainerstyl : {
        width: '362px',
        backgroundColor: '#fff',
        borderRadius: '4px'
      },
      cancelIcon: {
        color: '#8d8d8d',
        position: 'absolute',
        top: '19px',
        right: '19px',
        height: '18.7px',
        width: '18px',
        cursor: 'pointer'
      },
      cancelbtnstyl : {
        color: '#74797b'
      },
      deleteBtnstyl : {
        borderRadius: 2,
        backgroundColor: '#34b6b4'
      }
    }
    
    const actions = [
      <FlatButton
        label="Cancel"
        primary={true}
        style={DialogStyle.cancelbtnstyl}
        className="cancelBtn"
        onClick={this.handleModalClose} />,
      <FlatButton
        label="Delete"
        primary={true}
        onClick={that.handleRemoveBookmark.bind(that, that.state.bookmarkId)}
        style={DialogStyle.deleteBtnstyl}
        className="deleteBtn" />
    ];

    const CancelIcon = (props) => (
      <SvgIcon {...props}>
        <path d="M712.993036,23.3253012 L720.736289,15.2808193 C721.026024,14.9878313 721.026024,14.5128916 720.735855,14.2199036 C720.445687,13.9266988 719.975518,13.9266988 719.685349,14.2199036 L711.976795,22.005012 L704.268458,14.2199036 C703.978072,13.9266988 703.50812,13.9266988 703.217735,14.2199036 C702.927566,14.5128916 702.927566,14.9878313 703.217518,15.2808193 L710.960554,23.3253012 L703.217735,31.3697831 C702.927566,31.6627711 702.927566,32.1377108 703.217735,32.4306988 C703.507687,32.7234699 703.978072,32.7239036 704.268458,32.4306988 L711.976795,24.6455904 L719.685349,32.4306988 C719.975518,32.7239036 720.44612,32.7234699 720.735855,32.4306988 C721.026024,32.1377108 721.026024,31.6627711 720.735855,31.3697831 L712.993036,23.3253012 Z" id="ic_cancel" fill="#8d8d8d"></path>
      </SvgIcon>
    );

    return(<div><ul className="o-bookmark-list">
      {
        this.state.bookmarkList.map(function(bkmark) {
          const formatDateFrom_ms = new Date(bkmark.createdTimestamp);
          const date = formatDateFrom_ms.toLocaleDateString();
          const time = formatDateFrom_ms.toLocaleTimeString().replace(/(.*)\D\d+/, '$1');
          return <li
            className="o-bookmark-section"
            key={bkmark.id} >
            <a className="o-bookmark-content"
              onFocus= {that.onFocus.bind(that)}
              onKeyDown= {that.onLiBlur.bind(that)}
              data-uri={bkmark.uri}
              href="javascript:void(0)"
              onClick = {that.handleClick.bind(that, bkmark.uri)}
              onKeyPress={that.handleClick.bind(that, bkmark.uri)}>{bkmark.title}
              <div className="o-bookmark-date">
                <time value={bkmark.createdTimestamp}>{date + ' ' + time}</time>
              </div>
            </a>
            <a href="javascript:void(0);"
              onBlur={that.onBlur.bind(that)}
              className="remove"
              onClick= {that.handleModalOpen.bind(this, bkmark.id)}
              role="button">
            </a>
            </li>
        })
      }
    </ul>
    <Dialog
        title="Confirm Delete?"
        actions={actions}
        modal={false}
        open={that.state.modalOpen}
        onRequestClose={that.handleModalClose}
        contentStyle={DialogStyle.dialogContainerstyl}>
        <CancelIcon onClick={that.handleModalClose} viewBox="703 14 18 18.7" style={DialogStyle.cancelIcon}/> 
        This action cannot be undone.
    </Dialog>
     </div>     
    );
  }

  render() {
    if (this.state.bookmarkList.length) {
      return this.renderBookmarks();
    } else {
      return this.renderNoBookmarks();
    }
  }
}

BookmarkList.propTypes = {
  // intl: intlShape.isRequired,
  locale: PropTypes.string,
  clickBookmarkHandler: PropTypes.func,
  bookmarksArr: PropTypes.array.isRequired
};

export default BookmarkList;
