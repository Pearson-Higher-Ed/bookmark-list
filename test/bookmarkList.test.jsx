/**
PEARSON PROPRIETARY AND CONFIDENTIAL INFORMATION SUBJECT TO NDA
 *  Copyright Â© 2017 Pearson Education, Inc.
 *  All Rights Reserved.
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of Pearson Education, Inc.  The intellectual and technical concepts contained
 * herein are proprietary to Pearson Education, Inc. and may be covered by U.S. and Foreign Patents,
 * patent applications, and are protected by trade secret or copyright law.
 * Dissemination of this information, reproduction of this material, and copying or distribution of this software
 * is strictly forbidden unless prior written permission is obtained
 * from Pearson Education, Inc.
**/
/* eslint-disable */

import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { IntlProvider } from 'react-intl';
import Dialog from 'material-ui/Dialog';
import { InternationalSupport } from '@pearson-incubator/aquila-js-core';
import BookmarkList from '../src/js/BookmarkList';
import msgObject from '../translations';


 const bookmarksArr = [
  {
    "id": "1499682113569",
    "uri": "urn:pearson:manifestation:1c31193b-c5fe-4b17-881d-1f591351e82e",
    "data": {
      "baseUrl": "urn:pearson:manifestation:55a98cce-067a-4a73-b610-229959548eab"
    },
    "createdTimestamp": 1499682113569,
    "title": "Neurons and Nerves",
    "labels": "Neurons and Nerves"
  }
]

const clickBookmarkHandler = () => {
    console.log('clickBookmarkHandler');
  };


const intlObj = new InternationalSupport(msgObject, 'en');

  it('renders the renderNoBookmarks function', () => {
  	const component = renderer.create(
    <MuiThemeProvider>
    <IntlProvider locale={intlObj.getLocale()} messages={intlObj.getMessages()}>
    <BookmarkList
      clickBookmarkHandler={clickBookmarkHandler}
      removeBookmarkHandler={()=>{}}
      />
    </IntlProvider>
    </MuiThemeProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  // manually trigger the callback
  tree.children[0].children[0].props;
  tree.children[0].children[0].children[0];
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

 it('renders the renderBookmarks function', () => {
    const component = renderer.create(
    <MuiThemeProvider>
    <IntlProvider locale={intlObj.getLocale()} messages={intlObj.getMessages()}>
    <BookmarkList
      bookmarksArr={bookmarksArr} 
      clickBookmarkHandler={clickBookmarkHandler}
      removeBookmarkHandler={()=>{}}
    />
      </IntlProvider>
      </MuiThemeProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
   tree.children[0].children[0].children[0].props.onClick({preventDefault(){}, target:{parentNode:{classList:{add(){}}}}});
   tree.children[0].children[0].children[0].props.onFocus({preventDefault(){}, target:{parentNode:{classList:{add(){}}}}});
   tree.children[0].children[0].children[0].props.onKeyDown({preventDefault(){}, target:{parentNode:{classList:{remove(){}}}}, shiftKey:true,keyCode:9});
   tree.children[0].children[0].children[0].props.onKeyPress({preventDefault(){}, target:{parentNode:{classList:{add(){}}}}});
   tree.children[0].children[0].children[1].props.onClick({preventDefault(){}, target:{parentNode:{classList:{add(){}}}}});
   tree = component.toJSON();
   tree.children[0].children[0].children[0].props.onKeyUp({preventDefault(){}, which:37, keyCode:37, target:{parentNode:{classList:{remove(){}}}}});
   expect(tree).toMatchSnapshot();
});










