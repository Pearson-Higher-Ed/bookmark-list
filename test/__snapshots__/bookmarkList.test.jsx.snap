// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the renderBookmarks function 1`] = `
<div>
  <ul
    className="o-bookmark-list"
  >
    <li
      className="o-bookmark-section"
    >
      <a
        className="o-bookmark-content"
        data-uri="urn:pearson:manifestation:1c31193b-c5fe-4b17-881d-1f591351e82e"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        Neurons and Nerves
        <div
          className="o-bookmark-date"
        >
          <time
            value={1499682113569}
          >
            7/10/2017
          </time>
           
          <time
            value={1499682113569}
          >
            3:51 PM
          </time>
        </div>
      </a>
      <a
        aria-label="Remove Bookmarks"
        className="remove"
        onClick={[Function]}
        role="button"
        tabIndex="0"
      >
        
      </a>
    </li>
  </ul>
</div>
`;

exports[`renders the renderBookmarks function 2`] = `
<div>
  <ul
    className="o-bookmark-list"
  >
    <li
      className="o-bookmark-section"
    >
      <a
        className="o-bookmark-content"
        data-uri="urn:pearson:manifestation:1c31193b-c5fe-4b17-881d-1f591351e82e"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        Neurons and Nerves
        <div
          className="o-bookmark-date"
        >
          <time
            value={1499682113569}
          >
            7/10/2017
          </time>
           
          <time
            value={1499682113569}
          >
            3:51 PM
          </time>
        </div>
      </a>
      <a
        aria-label="Remove Bookmarks"
        className="remove"
        onClick={[Function]}
        role="button"
        tabIndex="0"
      >
        
      </a>
    </li>
  </ul>
</div>
`;

exports[`renders the renderNoBookmarks function 1`] = `
<div
  className="o-bookmark-empty-help"
>
  <div
    className="o-bookmark-empty-message"
    role="article"
    tabIndex="0"
  >
    <p>
      No Bookmarks Where Found.
    </p>
  </div>
</div>
`;

exports[`renders the renderNoBookmarks function 2`] = `
<div
  className="o-bookmark-empty-help"
>
  <div
    className="o-bookmark-empty-message"
    role="article"
    tabIndex="0"
  >
    <p>
      No Bookmarks Where Found.
    </p>
  </div>
</div>
`;
